### PR TITLE
feat: add 'autoSetup' option

### LINKED ISSUE #5094: Feature: would be great to have an option to disable autoSetup
## Description
since we init players manually there is no need to run the autosetup loop.

maybe its possible to implement with global setting `videojs.options.autoSetup = false;`
in the same tick as the video.js script gets excuted.

### PR DESCRIPTION
**Pull request by axten**
fixes #5094

## Description
this must be set globally with `videojs.options.autoSetup = false` in the same tick as videojs source is loaded to take effect

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [ ] Unit Tests updated or fixed
  - [x] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](http://jsbin.com/axedog/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by axten** [comment]
@gkatsev is something missing?

**Comment by gkatsev** [comment]
nope, last week as hackweek at work, so, I didn't really look at Video.js issues and PRs.
I'll take a look this week.

File: src/js/setup.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ let videojs;
 const autoSetup = function() {
 
   // Protect against breakage in non-browser environments.
-  if (!Dom.isReal()) {
+  if (!Dom.isReal() || !videojs.options.autoSetup) {
```
Comment (gkatsev) [inline src/js/setup.js]: while this would work, it still runs the initial timer and makes the comment above incorrect.

Comment (axten) [reply src/js/setup.js]: ok I will change this

**Comment by axten** [comment]
so changes are done... but now it's impossible to set this option, isn't it?

**Comment by axten** [comment]
Yes, confirm. I will revert changes...

**Comment by gkatsev** [comment]
@axten do you want this to go into Video.js 6 or will just having it in v7 be ok?

**Comment by gkatsev** [comment]
went ahead and ported it over.

**Pull request merged**